Sources/EditController.swift

```diff
@@ -319,9 +319,23 @@ extension EditController: AddToPlaylistViewControllerDelegate {
     func addToPlaylistViewController(_ addToPlaylistViewController: AddToPlaylistViewController,
                                      didSelectPlaylist playlist: VLCMLPlaylist) {
         let files = model.anyfiles
+
         for index in selectedCellIndexPaths where index.row < files.count {
-            if !playlist.appendMedia(withIdentifier: files[index.row].identifier()) {
-                assertionFailure("EditController: AddToPlaylistViewControllerDelegate: Failed to add item.")
+            if let mediaCollection = files[index.row] as? MediaCollectionModel {
+                guard let collectionFiles = mediaCollection.files() else {
+                    // unable to retreive files from a media collection
+                    continue
+                }
+
+                for file in collectionFiles {
+                    if !playlist.appendMedia(withIdentifier: file.identifier()) {
+                        assertionFailure("EditController: AddToPlaylistViewControllerDelegate: Failed to add item.")
+                    }
+                }
+            } else {
+                if !playlist.appendMedia(withIdentifier: files[index.row].identifier()) {
+                    assertionFailure("EditController: AddToPlaylistViewControllerDelegate: Failed to add item.")
+                }
```

Comment:
Tiny duplication, if it is a deal breaker, I can look into it later on if needed.

_I need to go back to medialibrary cell states coding._ 🤐